plugin/bufsize/bufsize.go

```diff
@@ -0,0 +1,31 @@
+// Package bufsize implements a plugin that modifies EDNS0 buffer size.
+package bufsize
+
+import (
+	"context"
+
+	"github.com/coredns/coredns/plugin"
+
+	"github.com/miekg/dns"
+)
+
+// Bufsize implements bufsize plugin.
+type Bufsize struct {
+	Next plugin.Handler
+	Size int
+}
+
+// ServeDNS implements the plugin.Handler interface.
+func (buf Bufsize) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg) (int, error) {
+	if option := r.IsEdns0(); option != nil {
+		option.SetUDPSize(uint16(buf.Size))
```

Comment:
does this do the right thing for the DO bit? It keep it, if it's there?

Reply:
yes, SetUDPSize() doesn't touch DO bit. It keeps original DO bit value.
https://github.com/miekg/dns/blob/master/edns.go#L127

Also I have tested and confirmed by pcap.